### DISCUSSION TITLE
Inferring type with `zod.input` when using `zod.coerce`

### DISCUSSION THREAD

**Discussion by niallkeys** [colinhacks/zod#2181] [Q&A] [answered]
```ts
const exampleObject = zod.object({
    date: zod.coerce.date()
});

export type exampleTypeInput = zod.input<typeof exampleObject>;
```

In this `exampleTypeInput` inferred for the attribute `date` has type `Date` Can this either be 'Any' or an option to add a type? Current issue means that we are unable to see this work with passing in a non `Date` value for example a `String`

**Comment by JacobWeisenburger** [comment]
I think this is being addressed here: https://github.com/colinhacks/zod/pull/1793

**Comment by colinhacks** [comment]
This is unlikely to be fixed in the near future unfortunately https://github.com/colinhacks/zod/issues/1760#issuecomment-1407816838

You may have to define some helpers like this:

```ts
z.any().transform(Number).pipe(z.number())
```

**Comment by JacobWeisenburger** [comment, marked as answer]
> ```ts
> z.any().transform(Number).pipe(z.number())
> ```

If you are going to do the above, then I would recommend doing it this way. Seems cleaner.

```ts
const schema = z.any().pipe( z.coerce.number() )
z.input<typeof schema> // any
z.output<typeof schema> // number
```

**Comment by colinhacks** [comment]
Nice, definitely better
